### ISSUE TITLE
extracting data from climate index maps

### ISSUE THREAD

**Issue by claudiahahn** [clarity-h2020/csis#163]
@ghilbrae, @LauraMTG, @luis-meteogrid  
Is there a way to extract data from the maps for the study area selected? 
I have the impression that was done initially as we tried to display the hazard table. Or am I mistaken? It would be great to somehow be able to get the value of the indices for the historical and future time periods, either displayed when moving over the map or extracted and made available in table format.  

@p-a-s-c-a-l: Not sure if it is the right place to post this.

**Comment by p-a-s-c-a-l** [comment]
Yes we had [this table](https://github.com/clarity-h2020/table-components) with aggregated values (only for heat) but then we couldn't cope with changes in the data structure and then eventually abandoned it in favour of a [simpler table](https://github.com/clarity-h2020/simple-table-component) which is fed by EMIKAT data only (HC: HW Events; EE: Population; HC-LE: TMRT, etc.; IA: Mortality Rate). So this new simple table works only with a limited set of layers coming from EMIKAT.

If you want to show / extract data for **any** layer (HC indices), this can become arbitrarily complex, not only for implementation but also for the user who wants to use this feature. Therefore we have to define clear requirements and give answer to the following questions:

1) Use case
What is the actual use case that shall be supported? What is the added value for the user?
Does 'extract' mean exporting (to a file), or just showing data for a single grid cell (on map) or all data for the whole study area (in table). Shall this information be made available in the report?

2) How to select the dataset to be shown / exported?
This should be as simple as possible, otherwise nobody is going to use it. ATM the user can select **one** scenario (combination of RCP, time period, event frequency) which controls which layers are available for selection in the map:
![grafik](https://user-images.githubusercontent.com/1788226/83733061-4d6e7a80-a64d-11ea-82ee-25fbc3dc9a6c.png)
In the future, we plan to allow selection of two scenarios for side-by-side comparison. 

Then, the user can select **one or more** layers:

![grafik](https://user-images.githubusercontent.com/1788226/83735281-75aba880-a650-11ea-94e6-b2905241c131.png)

Now, the question is: What's the actual dataset / layer we shall show the values for? All of them? Or just one? Which one? Who decides and how? If we want to show all values of all selected layers, how could such a user interface look like and how useful would that be?

3) Which values from the selected dataset(s) to show? 
There are similar requests from @DenoBeno for showing HC-LE and IA data coming from EMIKAT on the map: Click on the map or hover mouse over map and show the values, e.g. in a tooltip or popup. So far so good. But which values? If you want to support **arbitrary** data layers with arbitrary data structure, this can get complicated really fast. Consider the HC-LE layers which contains among T_MRT many other values. What to show on the map? Just T_MRT or all values available? How do we decide what is interesting for the user? Do we make this configurable or even user-selectable for each different layer (most complicated solution)? Or do we show all data (columns) for the current gird cell (KISS solution)? But how useful is this for the user if we show a long list of values including internal ones (gird id) and without units (this information is not available)?

4) How to access the data? This is a technical question that has to answered by developers.
Data from GeoServer and EMIKAT is accessed differently and has also a different structure (data formats.  
@therter how did the initial table component receive its data? WCS? E.g. how can we get the values of [this layer](https://geoserver.myclimateservice.eu/geoserver/europe/wms?service=WMS&version=1.1.0&request=GetMap&layers=europe%3ACI_consecutive-frost-days_rcp26_20410101-20701231_ensmean&bbox=2145642.726143716%2C982955.8095900388%2C6605432.868301096%2C5706496.981659953&width=725&height=768&srs=EPSG%3A3035&format=image%2Fpng)? The source is a GeoTIFF, so we can probably just show a single numeric value without any unit?

**Comment by RobAndGo** [comment]
1. I thought it could be good to have a numerical value "pop up" in a message box when the user moves their mouse over the map. This may be useful in case the colour scale makes it hard to see differences in the fields. I didn't think it would be necessary to have the values output to file nor included in the report. That is, this is simply an additional visual tool.

2. & 3. Once the variable or climate index is selected and thus shown, the value that is shown in the "pop up" is only that of the last (newest) field selected, in the case that multiple layers are shown. This would avoid having too much information pop up.

4. I was hoping that accessing data from GeoTIFF would not be that complex. In this case a simple value would be sufficient, as the units are shown in the description on the index which is (hopefully) somewhere accessible for the user.

**Comment by claudiahahn** [comment]
I agree with @RobAndGo. 

1. Having the pop-up showing the numerical value would help if the colorscale is not great. What I had in mind as well was the ability to show the user how an index will change in an easy way (right now one has to change the time period and select the layer again. Having two maps showing different time periods, as you have mentioned, will already help). To keep it "simple", showing the data for a single grid cell should be enough. If the area selected comprises more HC (12km x 12km) cells, one could define to use the cell with the largest area and mention this shortcoming? To make the data available maybe one could provide it as a csv by clicking on a button below the maps? Initially I thought of having the graphs as I have done for the reports for Agios Dimitrios etc. for EWRC 2019 (mentioned in an issue, which is closed since the time is probably too short)

2. Also for the last layer selected, but here one would need to not only select the value of the map shown but also the value of the same index for the other time periods and maybe one RCP (ideally plus the respective standard deviation value)

**Comment by p-a-s-c-a-l** [comment]
O.K. I think what @RobAndGo proposed is achievable once we know how to get the actual values from the GeoTIFF. Everything else may be supported later.
